src/seedu/addressbook/data/person/Address.java

```diff
@@ -52,14 +52,18 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        return block.getBlock() + street.getStreet() + unit.getUnit() + postal.getPostal();
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.block.getBlock().equals( ((Address) other).block.getBlock() )
```

Comment:
you can think of implementing a (or an!) equals method  in each class and use that here.

similarly the hashcode.